app/models/driver.rb

```diff
@@ -0,0 +1,40 @@
+
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+
+  # must provide a name
+  validates :name, presence: {message: "%{value} must be given"}
+
+  # must provide a vin
+  validates :vin, presence: {message: "%{value} must be given"}
+
+  def self.select_driver
+    return Driver.all.sample
+  end
+
+  def average_rating
+    sum = 0
+    counter = 0
+    trips.each do |trip|
+      if trip.rating != nil
+        sum += trip.rating
+        counter += 1
+      end
+    end
+    if sum != 0
```

Comment:
I think this should be `counter != 0` instead, because if `sum` is 0 then the average should be 0. If `counter` is 0 then we'll get an error because the next line will be dividing by 0.